pkg/sql/sem/builtins/builtins.go

```diff
@@ -1702,7 +1702,63 @@ var builtins = map[string]builtinDefinition{
 			tree.VolatilityImmutable,
 		)),
 
-	"similar_to_escape": makeBuiltin(defProps(),
+	"similar_escape": makeBuiltin(
+		tree.FunctionProperties{
+			NullableArgs: true,
+		},
+		tree.Overload{
+			Types:      tree.VariadicType{VarType: types.String},
```

Comment:
we shouldn't use VariadicType; we should define a 1 and two node variant.

Since `similar_escape` and similar_to_escape` are similar, we can make do this:

```
similarEscapeOverloads := []tree.Overload{
   ... put both overloads here ...
}

"similar_escape": makeBuiltin(
  tree.FunctionProperties{			NullableArgs: true,		},
  similarEscapeOverloads...
),
"similar_to_escape": makeBuiltin(
  tree.FunctionProperties{	},
  threeArgOverload,
  similarEscapeOverloads...
),
```